internal/domain/shortener/valueobject/original_url.go

```diff
@@ -0,0 +1,67 @@
+package valueobject
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type OriginalURL struct {
+	scheme string
+	host   string
+	port   int
+}
+
+func NewOriginalURL(originalURL string) (OriginalURL, error) {
+	scheme, hp, err := ParseOriginalURL(originalURL)
+	var port int
+
+	if len(hp) == 2 {
+		port, err = strconv.Atoi(hp[1])
+	}
+
+	if err != nil {
+		return OriginalURL{}, err
+	}
```

Comment:
Стоит перенести обработку ошибок как можно ближе к самому вызову.
```suggestion
	if len(hp) == 2 {
		port, err = strconv.Atoi(hp[1])
	         if err != nil {
		          return OriginalURL{}, err
	         }
	}
```